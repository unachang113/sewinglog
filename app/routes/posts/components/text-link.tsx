type Properties = {
  readonly path: string;
};

export function TextLink({path}: Properties) {
  return <a href={path}>Click me</a>;
}
