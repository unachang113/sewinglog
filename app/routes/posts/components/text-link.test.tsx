import {render} from '@testing-library/react';
import {TextLink} from './text-link';

describe('TextLink', () => {
  it('HTMLとスタイルが変わらないこと', () => {
    // テストの実装
    const {asFragment} = render(<TextLink path="https://example.com" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('pathで指定したURLがhref属性に設定されること', () => {
    const {getByRole} = render(<TextLink path="https://example.com" />);

    const link = getByRole('link');
    expect(link).toHaveAttribute('href', 'https://example.com');
  });
});
