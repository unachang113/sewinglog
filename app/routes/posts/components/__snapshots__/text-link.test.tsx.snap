// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextLink > HTMLとスタイルが変わらないこと 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    Click me
  </a>
</DocumentFragment>
`;
